scipy/stats/stats.py

```diff
@@ -2980,8 +2980,8 @@ def pearsonr(x, y):
 
     The p-value roughly indicates the probability of an uncorrelated system
     producing datasets that have a Pearson correlation at least as extreme
-    as the one computed from these datasets. The p-values are not entirely
-    reliable but are probably reasonable for datasets larger than 500 or so.
+    as the one computed from these datasets. The p-value is calculated using
+    the t-distribution as described in [1]_.
```

Comment:
add: "This distribution is exact under the null hypothesis if the data is normally distributed. Otherwise it is only an approximation, see [2]_ for a discussion on the reliability of the p-value in the non-normal case."

2 = C.J. Kowalski: On the Effects of Non-Normality on the Distribution of the Sample Product-Moment Correlation Coefficient.Journal of the Royal Statistical Society. Series C (Applied Statistics)
Vol. 21, No. 1 (1972), pp. 1-12